src/Evolution/Executables/GeneralizedHarmonic/EvolveGhSingleBlackHole.hpp

```diff
@@ -248,6 +252,7 @@ static const std::vector<void (*)()> charm_init_node_funcs{
     &domain::creators::time_dependence::register_derived_with_charm,
     &domain::FunctionsOfTime::register_derived_with_charm,
     &gh::BoundaryCorrections::register_derived_with_charm,
+    &control_system::size::register_derived_with_charm,
```

Comment:
So I understand, why must size register but not shape or other control systems?

Reply:
The `control_system::size::State`s are all derived classes from a base, so you have to register all the derived classes with charm. Just like functions of time, domain creators, time dependences, etc. The other control systems don't have base/derived classes so there's no need to register anything.